### ISSUE TITLE
Touchpad gesture to switch workspaces (added in 1.0.1-a.18) no longer works [1.0.2-b.1]

### ISSUE THREAD

**Issue by Arbifoxx** [zen-browser/desktop#3655]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

I am unable to switch workspaces by swiping with two fingers like in Arc anymore. It simply does nothing. I loved this functionality, and it was one of my favorite parts of this browser. The ability to switch workspaces allows for a seamless transition between work tabs and data and personal tabs and data, and keeps the two separate without having to switch profiles. Also, when this functionality was in Zen, there was a bug where if one swiped too hard or too fast, it would switch workspaces once and then again, skipping past the workspace that I wanted to switch too. This is especially annoying if there are only two workspaces, as it would switch to the other workstation and then back to the original workstation. Please bring this functionality back, it's the only thing keeping me on Arc and not Zen! (And animations too, just a note)

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.2-b.1

### What platform are you seeing the problem on?

macOS - aarch64

**Comment by Arbifoxx** [comment]
As a sidenote on workstpaces, do cookies between stations stay seperated? I believe that they do, but I don't know the level of seperation between workspaces.

**Comment by Arbifoxx** [comment]
Also, this may seem like a duplicate, but I checked other issues and the other ones only mention the sensitivity of switching. I did mention that, but the main issue here is that the gesture simply doesn't work.

**Comment by Arbifoxx** [comment]
It seems the last time this was working was 1.0.1-a.22

**Comment by signalno11** [comment]
WFM: macOS 12 x86, Zen 1.0.2-b1.

**Comment by signalno11** [comment]
> As a sidenote on workstpaces, do cookies between stations stay seperated? I believe that they do, but I don't know the level of seperation between workspaces.

Workspaces don't separate cookies at all if I understand. To achieve separation, assign a default container to the workspace.

**Comment by Arbifoxx** [comment]
Gotcha. Are you having this issue too? I'm not sure what WFM is.

**Comment by Arbifoxx** [comment]
I am going to try and find which commit changed this behavior, make a fork and maybe a PR. I'm not confident I can do that since I'm not experienced in this type of coding, but I'm going to try.

**Comment by signalno11** [comment]
> Gotcha. Are you having this issue too? I'm not sure what WFM is.

"WFM" means "Works for me."

Maybe try creating a fresh profile (about:profiles) and seeing if it works there?

**Comment by Arbifoxx** [comment]
Odd, and I will. The weird part to me then is the fact that it works on the alpha but not the beta. I am on Sonoma if that matters.

**Comment by Arbifoxx** [comment]
Still no. And interestingly enough, the workspaces button simply doesn't even appear when the new profile is created. I have to first close and reopen it. Also, the workspaces do still work, but the gesture doesn't.

**Comment by Arbifoxx** [comment]
This still has not been fixed. Should I remake this issue?

**Comment by neurokitti** [comment]
> This still has not been fixed. Should I remake this issue?

dont remake issues. not after only two weeks anyway

**Comment by neurokitti** [comment]
i fixed the swiping with this PR: https://github.com/zen-browser/desktop/pull/3289

not sure why you are having this issue as I have tested it and dont seem to be having the problem on my macbook or windows PC.
can you send a video of it not working so I can trouble shoot?

**Comment by ethnzhng** [comment]
I believe I found the issue: 

When the `Swipe between pages` setting in `Trackpad > More Gestures` of macOS system systems is set to `Off`, the workspace-switching gestures do not work in Zen. When this gesture is enabled, then it can be used to switch workspaces in Zen. In Arc (and possibly previous builds of Zen), the sidebar gesture works even if the system gesture is disabled, which should be the expected behavior as some users like myself use the swipe to change workspaces but not to navigate pages. This is reproducible in `1.0.2-b.5`

**Comment by HubertasVin** [comment]
I am a Linux user of Zen, and I can tell you that I have the gestures setting enabled for Gnome and the gestures do not work in Zen.

**Comment by HubertasVin** [comment]
> I believe I found the issue: 
> 
> When the `Swipe between pages` setting in `Trackpad > More Gestures` of macOS system systems is set to `Off`, the workspace-switching gestures do not work in Zen. When this gesture is enabled, then it can be used to switch workspaces in Zen. In Arc (and possibly previous builds of Zen), the sidebar gesture works even if the system gesture is disabled, which should be the expected behavior as some users like myself use the swipe to change workspaces but not to navigate pages. This is reproducible in `1.0.2-b.5`


I am a Linux user of Zen, and I can tell you that I have the gestures setting enabled for Gnome and the gestures do not work in Zen.

**Comment by GJRobert** [comment]
Not working in my Twilight 1.6t (macOS 13.1 Ventura) even if `zen.workspaces.swipe-actions` set to `true`.

I recall it working in Zen 1.0.2-b.2 or b.3 on my side (which may be different with other people).

**Comment by JSGRD22** [comment]
Are you still having this issue

**Comment by GJRobert** [comment]
Yes, still not working for me, Zen 1.7.1b.

**Comment by JSGRD22** [comment]
There is a similar bug where touchpad swipe does not work if too many tabs are opened i.e. it overflows and becomes a scroll box. Is that what you are experiencing, or does swiping on an empty space not work either?
Also try reproducing on a new profile.

**Comment by GJRobert** [comment]
Yes you are right, not working on overflown tabbar only, still works on spaces with few tabs open.

**Comment by Arbifoxx** [comment]
Didn’t know this issue was still active, I do have a video of it in action here: https://github.com/zen-browser/desktop/issues/4084

**Comment by Arbifoxx** [comment]
Having the swipe between pages set to two or three fingers fixes it for me though.

**Comment by JSGRD22** [comment]
> Yes you are right, not working on overflown tabbar only, still works on spaces with few tabs open.

In that case, we can consider the original issue resolved. The one relevant to the currently present bug is here: #4375

**Comment by theDanielJLewis** [comment]
Sorry, but with a brand new installation, new profile, only the 1password extension, I cannot use swipe gestures with my Apple touchpad on macOS. I'm in Zen Browser 1.7.2. Swiping doesn't go forward and backward with pages, and it doesn't swipe workspaces.
